# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../moneypenny/qt-stl-string.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
Nur in moneypenny-work/bin: moneypenny.
diff -ur moneypenny-beta4/moneypenny.kdevelop moneypenny-work/moneypenny.kdevelop
--- moneypenny-beta4/moneypenny.kdevelop	2005-05-22 20:16:10.000000000 +0200
+++ moneypenny-work/moneypenny.kdevelop	2005-06-02 15:43:45.000000000 +0200
@@ -105,7 +105,7 @@
       <envvars/>
     </run>
     <general>
-      <activedir>templates</activedir>
+      <activedir></activedir>
     </general>
     <make>
       <abortonerror>true</abortonerror>
Nur in moneypenny-work/: moneypenny.kdevelop.pcs.
Nur in moneypenny-work/: moneypenny.kdevses.
Nur in moneypenny-work/src: build.
diff -ur moneypenny-beta4/src/databaseaccess.cpp moneypenny-work/src/databaseaccess.cpp
--- moneypenny-beta4/src/databaseaccess.cpp	2005-05-21 15:31:01.000000000 +0200
+++ moneypenny-work/src/databaseaccess.cpp	2005-06-02 12:37:35.000000000 +0200
@@ -605,10 +605,10 @@
    // xgettext:no-c-format
    result+= tr("Print Date           : ") + date.toString(dateFormat) + "\n";
    result+= tr("Time                 : ") + date.toString(timeFormat) + "\n";
-   result+= tr("User                 : ")+ userid + "\n";
-   result+= tr("Key number           : ")+HBCI::String::num2string(keynumber) + "\n";
-   result+= tr("Key version          : ")+HBCI::String::num2string(keyversion) + "\n";
-   result+= tr("Customer system      : ")+sysid+"\n";
+   result+= tr("User                 : ") + userid.c_str() + "\n";
+   result+= tr("Key number           : ") + HBCI::String::num2string(keynumber).c_str() + "\n";
+   result+= tr("Key version          : ") + HBCI::String::num2string(keyversion).c_str() + "\n";
+   result+= tr("Customer system      : ") + sysid + "\n";
    result+="\n";
 
     result+=tr("My public key for electronic signature\n");
@@ -620,7 +620,7 @@
                           0,
                           ERROR_ADVISE_DONTKNOW,
                           "exponent too short");
-        result+= HoBIT_String::dumpHexString(exponent);
+        result+= HoBIT_String::dumpHexString(exponent).c_str();
 
         result+="\n";
         result+=tr("  Modulus\n");
@@ -630,11 +630,11 @@
                           0,
                           ERROR_ADVISE_DONTKNOW,
                           "exponent too short");
-        result+= HoBIT_String::dumpHexString(modulus);
+        result+= HoBIT_String::dumpHexString(modulus).c_str();
 
         result+="\n";
         result+=tr("  Hash value for my public key:\n");
-        result+= HoBIT_String::dumpHexString(myHash);
+        result+= HoBIT_String::dumpHexString(myHash).c_str();
 
         result+="\n";
         result+=tr("I confirm that my computer uses the above key for my electronic \nsignature.\n\n");
@@ -654,7 +654,7 @@
     return error;
   }
 
-  return result;
+  return result.latin1();
 }
 
 /** Returns the date of the last stored transaction.
@@ -699,7 +699,7 @@
    // First create list
     HoBIT_Bank * bank;
     list<HoBIT_Bank> banks;
-    QFile file(filename);
+    QFile file(filename.c_str());
     QFileInfo fileInfo(file);
     qDebug("Reading bankfile: %s", filename.c_str());
     if ( file.open( IO_ReadOnly ) ) {
Nur in moneypenny-work/src: databaseaccess.cpp~.
diff -ur moneypenny-beta4/src/guiaccountinformation.cpp moneypenny-work/src/guiaccountinformation.cpp
--- moneypenny-beta4/src/guiaccountinformation.cpp	2005-05-19 13:14:20.000000000 +0200
+++ moneypenny-work/src/guiaccountinformation.cpp	2005-06-02 13:43:41.000000000 +0200
@@ -87,9 +87,9 @@
       bankName = QString(bank.ref().name().c_str());
 
     // get Balances
-    noted = QString(account.ref().notedBalance().value().toReadableString());
-    booked = QString(account.ref().bookedBalance().value().toReadableString());
-    creditLine = QString(account.ref().creditLine().toReadableString());
+    noted = QString(account.ref().notedBalance().value().toReadableString().c_str());
+    booked = QString(account.ref().bookedBalance().value().toReadableString().c_str());
+    creditLine = QString(account.ref().creditLine().toReadableString().c_str());
 
     new QListViewItem(listViewAccount, tr("Credit Line:"), creditLine);
     new QListViewItem(listViewAccount, tr("Noted Balance:"), noted);
Nur in moneypenny-work/src: guiaccountinformation.cpp~.
diff -ur moneypenny-beta4/src/guiburn.cpp moneypenny-work/src/guiburn.cpp
--- moneypenny-beta4/src/guiburn.cpp	2005-05-14 18:57:11.000000000 +0200
+++ moneypenny-work/src/guiburn.cpp	2005-06-02 13:55:55.000000000 +0200
@@ -24,6 +24,8 @@
 #include <qprogressbar.h>
 #include <qstringlist.h>
 
+#include <stdlib.h>
+
 GuiBurn::GuiBurn(QWidget *parent, const char *name)
  : GuiBurnBase(parent, name)
 {
Nur in moneypenny-work/src: guiburn.cpp~.
diff -ur moneypenny-beta4/src/guidataexport.cpp moneypenny-work/src/guidataexport.cpp
--- moneypenny-beta4/src/guidataexport.cpp	2005-05-16 22:07:09.000000000 +0200
+++ moneypenny-work/src/guidataexport.cpp	2005-06-02 15:16:49.000000000 +0200
@@ -45,7 +45,7 @@
 
 if (path){
 _filepath = path.ascii();
-lineEditFile->setText(_filepath);
+lineEditFile->setText(_filepath.c_str());
 }
 
 }
@@ -102,5 +102,5 @@
 	_mountpoint = mp;
 	_filepath = _mountpoint + "/" + MONEYPENNY_DEFAULT_EXPORT_FILE;
 
-	lineEditFile->setText(_filepath);
+	lineEditFile->setText(_filepath.c_str());
 }
\ Kein Zeilenumbruch am Dateiende.
Nur in moneypenny-work/src: guidataexport.cpp~.
diff -ur moneypenny-beta4/src/guifindmedium.cpp moneypenny-work/src/guifindmedium.cpp
--- moneypenny-beta4/src/guifindmedium.cpp	2005-05-14 18:57:11.000000000 +0200
+++ moneypenny-work/src/guifindmedium.cpp	2005-06-02 13:55:13.000000000 +0200
@@ -34,6 +34,8 @@
 #include <qframe.h>
 #include <qcursor.h>
 
+#include <stdlib.h>
+
 GuiFindMedium::GuiFindMedium(QWidget * parent)
  : GuiFindMediumBase(parent)
 {
@@ -363,7 +365,8 @@
    <br>Then a file dialog pops up. Select your mountpoint in this dialog and click Ok."));
   if(USE_LIVE_CD_FEATURES)
     system("sudo xterm");
-  else system("xterm");
+  else
+    system("xterm");
   QString s = QFileDialog::getExistingDirectory(
                  "/mnt",
                  this,
Nur in moneypenny-work/src: guifindmedium.cpp~.
diff -ur moneypenny-beta4/src/guilogin.cpp moneypenny-work/src/guilogin.cpp
--- moneypenny-beta4/src/guilogin.cpp	2005-05-19 13:14:20.000000000 +0200
+++ moneypenny-work/src/guilogin.cpp	2005-06-02 13:41:51.000000000 +0200
@@ -151,7 +151,7 @@
 void GuiLogin::slotNewDescription(){
     QString newDescription = lineEditDescription->text();
     qDebug("New Description: %s", newDescription.latin1());
-    _hobit->changeAccountName(comboBoxAccounts->currentText(), newDescription);       
+    _hobit->changeAccountName(comboBoxAccounts->currentText().latin1(), newDescription.latin1());
     buttonEdit->hide();
 }
 
Nur in moneypenny-work/src: guilogin.cpp~.
diff -ur moneypenny-beta4/src/guimain.cpp moneypenny-work/src/guimain.cpp
--- moneypenny-beta4/src/guimain.cpp	2005-05-22 20:16:10.000000000 +0200
+++ moneypenny-work/src/guimain.cpp	2005-06-02 13:55:08.000000000 +0200
@@ -641,7 +641,7 @@
 {
   if (_currentAccount != "")
   {
-    Pointer<Account> acc = _hobit->findAccount(_currentAccount);
+    Pointer<Account> acc = _hobit->findAccount(_currentAccount.latin1());
     if (acc.isValid())
     {
       QString balance, n;
@@ -651,7 +651,7 @@
       // n.setNum(val%100);
       // balance += n + " ";
       //balance += (*_accit).ref().bookedBalance().value().getCurrency().c_str();
-      balance = acc.ref().bookedBalance().value().toReadableString();
+      balance = acc.ref().bookedBalance().value().toReadableString().c_str();
       QString text = tr("Your Balance: ") + balance + " ";
       textLabelBalances->setText(text);
     }
@@ -1334,7 +1334,7 @@
   void GuiMain::slotExport(QTable* table){
    _currentWidget = widgetStack->visibleWidget();
    _guidataexport->setTableToExport(table);
-   _guidataexport->setMountPoint(_MountPoint);
+   _guidataexport->setMountPoint(_MountPoint.latin1());
 
     widgetStack->raiseWidget(EXPORT_WIDGET);
  }
Nur in moneypenny-work/src: guimain.cpp~.
diff -ur moneypenny-beta4/src/guitransactioninfo.cpp moneypenny-work/src/guitransactioninfo.cpp
--- moneypenny-beta4/src/guitransactioninfo.cpp	2005-05-16 22:07:10.000000000 +0200
+++ moneypenny-work/src/guitransactioninfo.cpp	2005-06-02 13:58:07.000000000 +0200
@@ -47,20 +47,20 @@
    clearAll();
 
    // own account
-   textLabelAccountIdFrom->setText(sto.ref().accountId());
-   textLabelBankIdFrom->setText(sto.ref().bankCode());
+   textLabelAccountIdFrom->setText(sto.ref().accountId().c_str());
+   textLabelBankIdFrom->setText(sto.ref().bankCode().c_str());
    textLabelCountryFrom->setText(QString::number(sto.ref().country()));
    
    // other account
-   textLabelOwnerTo->setText(sto.ref().otherName());
-   textLabelAccountIdTo->setText(sto.ref().otherAccountId());
-   textLabelBankIdTo->setText(sto.ref().otherBankId());
+   textLabelOwnerTo->setText(sto.ref().otherName().c_str());
+   textLabelAccountIdTo->setText(sto.ref().otherAccountId().c_str());
+   textLabelBankIdTo->setText(sto.ref().otherBankId().c_str());
    textLabelCountryTo->setText(QString::number(sto.ref().otherCountry()));
    
    
-   textLabelAmount->setText(sto.ref().value().toReadableString());
+   textLabelAmount->setText(sto.ref().value().toReadableString().c_str());
    label3->setText(tr("order number"));
-    QString jobid = sto.ref().jobId();
+   QString jobid = sto.ref().jobId().c_str();
    textLabelThird->setText(QString::number(jobid.toInt()));
    
    Date d = sto.ref().firstDate();
@@ -95,14 +95,14 @@
    clearAll();
    
     // own account
-   textLabelAccountIdFrom->setText(trc.ref().ouraccountId());
-   textLabelBankIdFrom->setText(trc.ref().ourbankcode());
+   textLabelAccountIdFrom->setText(trc.ref().ouraccountId().c_str());
+   textLabelBankIdFrom->setText(trc.ref().ourbankcode().c_str());
  //  textLabelCountryFrom->setText(QString::number(sto.ref().ourcountry()));
    
    // other account
-   textLabelOwnerTo->setText(trc.ref().othername());
-   textLabelAccountIdTo->setText(trc.ref().otheraccountId());
-   textLabelBankIdTo->setText(trc.ref().otherbankcode());
+   textLabelOwnerTo->setText(trc.ref().othername().c_str());
+   textLabelAccountIdTo->setText(trc.ref().otheraccountId().c_str());
+   textLabelBankIdTo->setText(trc.ref().otherbankcode().c_str());
 //   textLabelCountryTo->setText(QString::number(sto.ref().othercountry()));
 
     list<string> purposes = trc.ref().purposes();
@@ -130,8 +130,8 @@
    label3->setText(tr("valuta date"));
    textLabelFirst->setText(date3.toString());
    
-   textLabelPrimanota->setText(trc.ref().primanota());
-   textLabelAmount->setText(trc.ref().value().toReadableString());
+   textLabelPrimanota->setText(trc.ref().primanota().c_str());
+   textLabelAmount->setText(trc.ref().value().toReadableString().c_str());
    
 }
 
Nur in moneypenny-work/src: guitransactioninfo.cpp~.
diff -ur moneypenny-beta4/src/guitransferform.cpp moneypenny-work/src/guitransferform.cpp
--- moneypenny-beta4/src/guitransferform.cpp	2005-05-16 22:07:10.000000000 +0200
+++ moneypenny-work/src/guitransferform.cpp	2005-06-02 15:14:48.000000000 +0200
@@ -121,17 +121,22 @@
      // text for show all inputs of form, used in messagebox
      QString text =    tr("You are about to create the following transfer template: ") +"\n\n";
      text += tr("From Account:      ") ;
-     text +=  ourAccount + "\n";
-     text +=  tr("To Account: ");
-     text += accountId+ "\n";
+     text += ourAccount.c_str();
+     text += "\n";
+     text += tr("To Account: ");
+     text += accountId.c_str();
+     text += "\n";
      text += tr("At Bank:              ");
-     text += bankCode+ "\n";
+     text += bankCode.c_str();
+     text += "\n";
      text += tr("Amount:                 ");
-     text += value;
+     text += value.c_str();
      text += tr(" EUR")+ "\n\n";
      text += tr("\n\nPurpose:");
-     text += purpose1+ "\n";
-     text += purpose2 + "\n";
+     text += purpose1.c_str();
+     text += "\n";
+     text += purpose2.c_str();
+     text += "\n";
      
      // messagebox opens for show input of form and ask for confirm, alternatively decline
      bool no = QMessageBox::question( this,
Nur in moneypenny-work/src: guitransferform.cpp~.
diff -ur moneypenny-beta4/src/guitransfertemplates.cpp moneypenny-work/src/guitransfertemplates.cpp
--- moneypenny-beta4/src/guitransfertemplates.cpp	2005-05-14 18:57:11.000000000 +0200
+++ moneypenny-work/src/guitransfertemplates.cpp	2005-06-02 13:48:08.000000000 +0200
@@ -96,16 +96,14 @@
 {
 QListViewItem* item; 
 item = listViewTemplates->selectedItem();
-QString name =  item->text(0);
-QString account =  item->text(1);
-QString bank =  item->text(2);
-QString value =  item->text(3);
-QString purpose1 =  item->text(4);
-QString purpose2 =  item->text(5);
+string name =  item->text(0).latin1();
+string account =  item->text(1).latin1();
+string bank =  item->text(2).latin1();
+string value =  item->text(3).latin1();
+string purpose1 =  item->text(4).latin1();
+string purpose2 =  item->text(5).latin1();
 
-
-
-_hobit->deleteTransferTemplate(_main->getCurrentAccount(), name, account, bank, value, purpose1, purpose2);
+_hobit->deleteTransferTemplate(_main->getCurrentAccount().latin1(), name, account, bank, value, purpose1, purpose2);
 refresh();
 }
 
@@ -120,7 +118,7 @@
   
   _templates = _hobit->getTransferTemplates();
   list<Pointer<HoBIT_TransferTemplate> >::iterator it;
-  string currentAccount = _main->getCurrentAccount();
+  string currentAccount = _main->getCurrentAccount().latin1();
   
   listViewTemplates->clear();
   for(it = _templates.begin(); it != _templates.end(); it++){
@@ -139,7 +137,7 @@
   			name.c_str(),
 			account.c_str(),
 			bank.c_str(),
-			value.toReadableString(),
+			value.toReadableString().c_str(),
 			purpose.c_str(),
 			purpose2.c_str());
   }			
Nur in moneypenny-work/src: guitransfertemplates.cpp~.
diff -ur moneypenny-beta4/src/hobit.cpp moneypenny-work/src/hobit.cpp
--- moneypenny-beta4/src/hobit.cpp	2005-05-21 19:01:56.000000000 +0200
+++ moneypenny-work/src/hobit.cpp	2005-06-02 13:30:14.000000000 +0200
@@ -211,7 +211,9 @@
       return false;
     }
     cerr << "Loading Environment....\n";
-    Error err = _api.ref().loadEnvironment(_Crypt->getConfigDataDir() + "/" + _openhbcifile, _Crypt->getConfigDataDir() + "/" + _aqmoneyfile);
+
+    string cdir = (_Crypt->getConfigDataDir() + "/").latin1();
+    Error err = _api.ref().loadEnvironment( cdir + _openhbcifile, cdir + _aqmoneyfile);
     if (!err.isOk()) {
       DBG_ERROR(0, "Error loading environment (%s)",
                 err.errorString().c_str());
@@ -219,7 +221,7 @@
     return false;
    }
   
-   err = _Database->readTransferTemplates(_Crypt->getConfigDataDir() + "/" + _templatefile);
+   err = _Database->readTransferTemplates( cdir + _templatefile);
    if (!err.isOk()) {
    DBG_ERROR(0, "Error loading templates (%s)",
              err.errorString().c_str());
@@ -237,7 +239,9 @@
     
     // check if saving was possible
     cerr << "Writing environment.\n";
-    Error err = _api.ref().saveEnvironment(_Crypt->getConfigDataDir() + "/" + _openhbcifile, _Crypt->getConfigDataDir() + "/" + _aqmoneyfile);
+
+    string cdir = (_Crypt->getConfigDataDir() + "/").latin1();
+    Error err = _api.ref().saveEnvironment(cdir + _openhbcifile, cdir + _aqmoneyfile);
     if (!err.isOk()) {
       DBG_ERROR(0, "Error saving environment (%s)",
                 err.errorString().c_str());
@@ -246,7 +250,7 @@
   
     // check if saving was possible
     cerr << "Writing Templates.\n";
-    err = _Database->saveTransferTemplates(_Crypt->getConfigDataDir() + "/" + _templatefile);
+    err = _Database->saveTransferTemplates(cdir + _templatefile);
     if (!err.isOk()) {
       DBG_ERROR(0, "Error saving templates (%s)",
                 err.errorString().c_str());
@@ -494,7 +498,7 @@
   list<HoBIT_Bank > HoBIT::listOfAllKnownBanks(){
       if( _banklist.empty()){  
          QString filename = MP_DIR()  + QString(MONEYPENNY_BANKDATA_DIR) + "/"+ QString(MONEYPENNY_ALL_BANKDATA_FILE);
-         _banklist = _Database->readBankFile(filename); 
+         _banklist = _Database->readBankFile(filename.latin1()); 
       } 
       return _banklist;   
 }
@@ -506,7 +510,7 @@
   list<HoBIT_Bank> HoBIT::listOfKnownHbciBanks(){
      if(_hbcibanklist.empty()){
        QString filename = MP_DIR()  + QString(MONEYPENNY_BANKDATA_DIR) + "/"+ QString(MONEYPENNY_HBCI_BANKDATA_FILE);
-       _hbcibanklist = _Database->readBankFile(filename); 
+       _hbcibanklist = _Database->readBankFile(filename.latin1()); 
      }
      return _hbcibanklist;     
 }
@@ -984,13 +988,13 @@
  */
 void HoBIT::removeUncryptedEnvironment()
 {
-   QString rm = "rm " + _Crypt->getConfigDataDir() + "/" + _openhbcifile;
+   QString rm = "rm " + _Crypt->getConfigDataDir() + "/" + _openhbcifile.c_str();
    system(rm.latin1());
    
-   rm = "rm " + _Crypt->getConfigDataDir() + "/" + _aqmoneyfile;
+   rm = "rm " + _Crypt->getConfigDataDir() + "/" + _aqmoneyfile.c_str();
    system(rm.latin1());
    
-   rm = "rm " + _Crypt->getConfigDataDir() + "/" + _templatefile;
+   rm = "rm " + _Crypt->getConfigDataDir() + "/" + _templatefile.c_str();
    system(rm.latin1());
    
 }
@@ -1067,7 +1071,7 @@
     
    // now setting directories and file names
    //top directory is crypt::_ConfigDataDir 
-   _api.ref().setAapplicationDataDir(_Crypt->getConfigDataDir() + "/" + _hbcidatadir); //take care of Aa !!
+   _api.ref().setAapplicationDataDir((_Crypt->getConfigDataDir() + "/").latin1() + _hbcidatadir); //take care of Aa !!
   
    // creating JobFactories
    _AccountJobFactory = new AccountJobFactory(_api.ptr());
Nur in moneypenny-work/src: hobit.cpp~.
diff -ur moneypenny-beta4/src/hobit_crypt.cpp moneypenny-work/src/hobit_crypt.cpp
--- moneypenny-beta4/src/hobit_crypt.cpp	2005-05-22 10:00:31.000000000 +0200
+++ moneypenny-work/src/hobit_crypt.cpp	2005-06-02 13:41:11.000000000 +0200
@@ -179,22 +179,22 @@
 {
      FILE * tarFile, * cryptFile;
      //compress specified files and write them to the pipe on stdout (--file=- is stdout)
-     string tarcmd = "tar --create --gzip --file=- " + _ConfigDataDir + "/" + MONEYPENNY_HBCIDATA_DIR + // file =- means: use stdout for output
-     " " +  _ConfigDataDir + "/" + MONEYPENNY_AQMONEY_FILE +
+     QString tarcmd = "tar --create --gzip --file=- " + _ConfigDataDir + "/" + MONEYPENNY_HBCIDATA_DIR + // file =- means: use stdout for output
+     " " + _ConfigDataDir + "/" + MONEYPENNY_AQMONEY_FILE +
      " " + _ConfigDataDir + "/" + MONEYPENNY_OPENHBCI_FILE +
      " " + _ConfigDataDir + "/" + MONEYPENNY_TEMPLATE_FILE;
-     cerr << "tar command: " << tarcmd.c_str() << "\n";
+     cerr << "tar command: " << tarcmd.latin1() << "\n";
              
     //create the pipe for tar's output and start tar
-     tarFile = popen(tarcmd.c_str(),"r");
+     tarFile = popen(tarcmd.latin1(),"r");
      if(!tarFile){
              cerr<<"encryptConfigDataDir: Failed to execute tar command";
              return -1;
      }
        
      //build outfile string
-     string cryptFileName = _CryptDir + "/" + MONEYPENNY_ENCRYPTED_CONF_FILE;
-     cryptFile = fopen(cryptFileName.c_str(), "wb");
+     QString cryptFileName = _CryptDir + "/" + MONEYPENNY_ENCRYPTED_CONF_FILE;
+     cryptFile = fopen(cryptFileName.latin1(), "wb");
      if(!cryptFile){
        cerr<<"encryptConfigDataDir: Failed to open cryptFile for writing: " << cryptFileName << endl;
        return -2;
@@ -246,8 +246,8 @@
  */
 int HoBIT_Crypt::loadInitVector()
 {	
-	string ivpath = _CryptDir + "/iv.dat";
-	ifstream ivin(ivpath.c_str(),ios_base::binary);
+	QString ivpath = _CryptDir + "/iv.dat";
+	ifstream ivin(ivpath.latin1(),ios_base::binary);
 	if(!ivin){
 		cerr << "Failed to open iv.dat file for data decryption" << endl;
 		return -1;
@@ -277,18 +277,18 @@
 {
     	
 //store new init-vector before using it
-	string ivpath = _CryptDir + "/iv.dat";
+	QString ivpath = _CryptDir + "/iv.dat";
 	
 #ifdef DEBUG_CRYPT 
-	if(GWEN_Directory_GetPath(ivpath.c_str(),GWEN_PATH_FLAGS_PATHMUSTEXIST) == 0 ){
+	if(GWEN_Directory_GetPath(ivpath.latin1(),GWEN_PATH_FLAGS_PATHMUSTEXIST) == 0 ){
 		cerr << "deleting old iv.dat file" << ivpath << endl;
 	} else {
 		cerr << "iv.dat file does not exist! Creating new one" << endl;
 	}
 #endif
 
-	string newivpath = _CryptDir + "/iv.dat";
-	ofstream ivout(newivpath.c_str(),ios_base::binary);
+	QString newivpath = _CryptDir + "/iv.dat";
+	ofstream ivout(newivpath.latin1(),ios_base::binary);
 	if(!ivout){
 		cerr << "Failed to open iv.dat file for data decryption" << endl;
 		return -1;
@@ -310,7 +310,7 @@
 	if(filepath.empty()){
 		fp = _CryptDir + "/" + MONEYPENNY_PASSWORD_FILE;
 	} else {
-		fp = filepath;
+		fp = filepath.c_str();
 	}
 	cerr << "Password path:" << fp << endl;
 	ofstream ofmd(fp.latin1(),ios_base::binary);
@@ -337,8 +337,9 @@
 {
  	QString fileName;
 	if(filepath.empty())
-	  fileName = _CryptDir + "/" + MONEYPENNY_PASSWORD_FILE;
-        else fileName = filepath;
+		fileName = _CryptDir + "/" + MONEYPENNY_PASSWORD_FILE;
+        else
+		fileName = filepath.c_str();
         qDebug("Searching passwordfile: %s", filepath.c_str());
 	
 	ifstream stream(fileName.latin1(),ios_base::binary);
Nur in moneypenny-work/src: hobit_crypt.cpp~.
diff -ur moneypenny-beta4/src/hobit_tlv.cpp moneypenny-work/src/hobit_tlv.cpp
--- moneypenny-beta4/src/hobit_tlv.cpp	2005-04-17 13:43:11.000000000 +0200
+++ moneypenny-work/src/hobit_tlv.cpp	2005-06-02 15:10:45.000000000 +0200
@@ -117,10 +117,10 @@
   /** Create a readable string, format value as string
   */
   QString HoBIT_TLV::debugString(){   
-   return QString("HoBIT_TLV::debugString: Tag: " 
-                  + string((char ) _Tag, 1)  
-                  +"Length: " + QString::number(_Length) 
-                  +"Value: " + string(value(), _Length));
+   return QString( QString("HoBIT_TLV::debugString: Tag: ")
+                  + string((char) _Tag, 1).c_str()
+                  + "Length: " + QString::number(_Length)
+                  + "Value: " + string(value(), _Length).c_str());
   }
   
   /** Create a readable string, format value as Hex numbers
@@ -130,8 +130,8 @@
     s0.setNum(_Tag, 16);
     return QString("HoBIT_TLV::debugString: Tag: " 
                   + s0  
-                  +"Length: " + QString::number(_Length) 
-                  +"Value :" + HoBIT_String::dumpHexString(value(), _Length));
+                  + "Length: " + QString::number(_Length)
+                  + "Value :" + HoBIT_String::dumpHexString(value(), _Length).c_str());
  }
 
  /**Read the next TLV record form file.Returns number of bytes read.
Nur in moneypenny-work/src: hobit_tlv.cpp~.
diff -ur moneypenny-beta4/src/listviewitems.cpp moneypenny-work/src/listviewitems.cpp
--- moneypenny-beta4/src/listviewitems.cpp	2005-04-07 16:43:29.000000000 +0200
+++ moneypenny-work/src/listviewitems.cpp	2005-06-02 13:35:08.000000000 +0200
@@ -56,7 +56,7 @@
   QString result;
   Date d = _DatedTransfer.ref().date();
   QDate date(d.year(), d.month(), d.day());
-  string value = QString::number(_DatedTransfer.ref().value().getValue(),'f',2); // 2 digits after . "-999.99"  
+  string value = (QString::number(_DatedTransfer.ref().value().getValue(),'f',2)).latin1(); // 2 digits after . "-999.99"  
   string ouraccId = _DatedTransfer.ref().accountId();
   string ourbankCode = _DatedTransfer.ref().bankCode();
   string otheraccId = _DatedTransfer.ref().otherAccountId();
@@ -180,7 +180,7 @@
   QString result;
   Date d = _Transaction.ref().date();
   QDate date(d.year(), d.month(), d.day());
-  string value = QString::number(_Transaction.ref().value().getValue(),'f',2); // 2 digits after . "-999.99"  
+  string value = (QString::number(_Transaction.ref().value().getValue(),'f',2)).latin1(); // 2 digits after . "-999.99"  
   string ouraccId = _Transaction.ref().ouraccountId();
   string ourbankCode = _Transaction.ref().ourbankcode();
   string otheraccId = _Transaction.ref().otheraccountId();
@@ -282,7 +282,7 @@
   QString weekly(QString::number(_StandingOrder.ref().weekly()));
   string yearly = _StandingOrder.ref().yearly();
   
-  string value = QString::number(_StandingOrder.ref().value().getValue(),'f',2); // 2 digits after . "-999.99"  
+  string value = (QString::number(_StandingOrder.ref().value().getValue(),'f',2)).latin1(); // 2 digits after . "-999.99"  
   string ouraccId = _StandingOrder.ref().accountId();
   string ourbankCode = _StandingOrder.ref().bankCode();
   string otheraccId = _StandingOrder.ref().otherAccountId();
Nur in moneypenny-work/src: listviewitems.cpp~.
Nur in moneypenny-work/src: Makefile.
Nur in moneypenny-work/: tags.
